src/test/java/org/breedinginsight/brapi/v1/controller/BrapiObservationVariablesControllerIntegrationTest.java

```diff
@@ -0,0 +1,343 @@
+
+    @Inject
+    private DSLContext dsl;
+    @Inject
+    private ProgramDao programDao;
+
+    private List<Trait> programTraits;
+    private List<Trait> otherProgramTraits;
+    private String observationVariableDbId;
+
+    private ProgramEntity validProgram;
+    private ProgramEntity otherValidProgram;
+
+    @BeforeAll
+    @SneakyThrows
+    public void setup() {
+
+        // Insert our traits into the db
+        fp = FannyPack.fill("src/test/resources/sql/BrapiObservationVariablesControllerIntegrationTest.sql");
+
+        // Insert program
+        dsl.execute(fp.get("InsertProgram"));
+
+        // Insert program observation level
+        dsl.execute(fp.get("InsertProgramObservationLevel"));
+
+        // Insert program ontology sql
+        dsl.execute(fp.get("InsertProgramOntology"));
+        dsl.execute(fp.get("InsertTestProgramUser"));
+        dsl.execute(fp.get("InsertOtherTestProgramUser"));
+
+        // Retrieve our new data
+        validProgram = programDao.findAll().get(0);
+
+        // Insert other program
+        dsl.execute(fp.get("InsertOtherProgram"));
+        dsl.execute(fp.get("InsertOtherProgramObservationLevel"));
+        dsl.execute(fp.get("InsertOtherProgramOntology"));
+        dsl.execute(fp.get("InsertOtherTestOtherProgramUser"));
+
+        otherValidProgram = programDao.fetchByName("Other Test Program").get(0);
+
+        // Add species to BrAPI server
+        fp = FannyPack.fill("src/test/resources/sql/brapi/species.sql");
+        super.getBrapiDsl().execute(fp.get("InsertSpecies"));
+    }
+
+    Trait buildTestTrait(String name, String traitClass) {
+        Method method = Method.builder()
+                .methodName(name + " method")
+                .description(name + " method description")
+                .methodClass("Observation")
+                .build();
+
+        Scale scale = Scale.builder()
+                .scaleName(name + " scale")
+                .dataType(DataType.NUMERICAL)
+                .build();
+
+        ProgramObservationLevel level = ProgramObservationLevel.builder()
+                .name("Plant")
+                .build();
+
+        Trait trait = Trait.builder()
+                .traitName(name)
+                .description(name + " description")
+                .method(method)
+                .scale(scale)
+                .programObservationLevel(level)
+                .traitClass(traitClass)
+                .active(true)
+                .build();
+        return trait;
+    }
+
+    void checkTraits(Trait expected, JsonObject actual) {
+        JsonObject trait = actual.getAsJsonObject("trait");
+        assertEquals(expected.getTraitName(), actual.get("observationVariableName").getAsString(), "Wrong name");
+        assertEquals(expected.getDescription(), trait.get("description").getAsString(), "Wrong description");
+        //assertEquals(validLocation.getCountry().getName(), country.get("name").getAsString(), "Wrong country name");
```

Comment:
That is a copy/paste left over but I was debating doing a more thorough validation of the trait and decided not to for now in the interest of time and because some things will change once those field book workarounds are removed. If there are no objections I can just remove that comment and leave the minimal validation as-is for now.

Reply:
Sounds good

Reply:
Pushed changes.